src/empty_stubs.cpp

```diff
@@ -1,125 +1,502 @@
 #include "duckdb_odbc.hpp"
+#include "handle_functions.hpp"
+#include "statement_functions.hpp"
+#include "widechar.hpp"
 
-#include <iostream>
-
-//! ODBC stub functions not implemented yet,
+//! The following ODBC stub functions aren't implemented yet,
 //! when implementing the function must be moved to the proper source file
-//! Using std::cout instead of throw execptions because of MVSC's warning C4297
 
-SQLRETURN SQL_API SQLNativeSql(SQLHDBC connection_handle, SQLCHAR *in_statement_text, SQLINTEGER text_length1,
-                               SQLCHAR *out_statement_text, SQLINTEGER buffer_length, SQLINTEGER *text_length2_ptr) {
-	std::cout << "***** SQLNativeSql" << std::endl;
-	return SQL_ERROR;
-}
+// ============================================================================
+// Functions returning valid empty result sets (i.e., SELECT ... WHERE 1 < 0)
+// ============================================================================
 
-SQLRETURN SQL_API SQLNativeSqlW(SQLHDBC connection_handle, SQLWCHAR *in_statement_text, SQLINTEGER text_length1,
-                                SQLWCHAR *out_statement_text, SQLINTEGER buffer_length, SQLINTEGER *text_length2_ptr) {
-	std::cout << "***** SQLNativeSqlW" << std::endl;
-	return SQL_ERROR;
-}
+// -----------------------------------------------------------------------------
+// SQLPrimaryKeys and SQLPrimaryKeysW
+// -----------------------------------------------------------------------------
 
-SQLRETURN SQL_API SQLBrowseConnect(SQLHDBC connection_handle, SQLCHAR *in_connection_string, SQLSMALLINT string_length1,
-                                   SQLCHAR *out_connection_string, SQLSMALLINT buffer_length,
-                                   SQLSMALLINT *string_length2_ptr) {
-	std::cout << "***** SQLBrowseConnect" << std::endl;
-	return SQL_ERROR;
+static SQLRETURN PrimaryKeysInternal(SQLHSTMT statement_handle, SQLCHAR *catalog_name, SQLSMALLINT name_length1,
+                                     SQLCHAR *schema_name, SQLSMALLINT name_length2, SQLCHAR *table_name,
+                                     SQLSMALLINT name_length3) {
+	const std::string query = R"#(
+SELECT
+	CAST('' AS VARCHAR) AS "TABLE_CAT",
+	CAST('' AS VARCHAR) AS "TABLE_SCHEM",
+	CAST('' AS VARCHAR) AS "TABLE_NAME",
+	CAST('' AS VARCHAR) AS "COLUMN_NAME",
+	CAST(0 AS SMALLINT) AS "KEY_SEQ",
+	CAST('' AS VARCHAR) AS "PK_NAME"
+WHERE 1 < 0
+)#";
+	duckdb::OdbcHandleStmt *stmt = nullptr;
+	const SQLRETURN ret = ConvertHSTMT(statement_handle, stmt);
+	if (ret != SQL_SUCCESS) {
+		return ret;
+	}
+
+	return duckdb::ExecDirectStmt(stmt, duckdb::OdbcUtils::ConvertStringToSQLCHAR(query), query.size());
```

Comment:
Can we add a cast to `SQLINTEGER` here (and in other calls to `ExecDirectStmt`) for the query size, otherwise it triggers [C4267 warning](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4267?view=msvc-170) on MSVC.